apps/backend/src/modules/auth/auth.service.ts

```diff
@@ -25,36 +28,56 @@ class AuthService {
 		this.encrypt = encrypt;
 	}
 
+	public async checkResetPasswordExp(
+		body: ResetPasswordLinkDto,
+	): Promise<boolean> {
+		try {
+			await token.decode(body.link);
+
+			return true;
+		} catch (error) {
+			if (error instanceof JWTExpired) {
+				throw new AuthError({
+					message: ErrorMessage.RESET_PASSWORD_LINK_EXPIRED,
+				});
+			}
+
+			throw error;
+		}
+	}
+
 	public async forgotPassword(payload: EmailDto): Promise<boolean> {
 		const { email: targetEmail } = payload;
 
 		const user = await this.userService.findByEmail(targetEmail);
 
 		if (!user) {
 			throw new AuthError({
-				message: ErrorMessage.INCORRECT_CREDENTIALS,
+				message: ErrorMessage.EMAIL_NOT_FOUND,
 				status: HTTPCode.UNAUTHORIZED,
 			});
 		}
 
 		const userDetails = user.toObject();
 
 		const jwtToken = await token.createToken({
-			userId: userDetails.id,
+			expirationTime: config.ENV.DURATIONS.PASSWORD_RESET_LINK,
```

Comment:
expirationTime should be sent within a constructor

Reply:
Haven't we came to a conclusion that we should pass the duration as an argument to token creation method during one of the daily meetings?

Reply:
In a constructor for auth service, not for a token module